build-tools/automation/yaml-templates/run-integrated-regression-tests.yaml

```diff
@@ -51,160 +54,163 @@ steps:
-  parameters:
-    nunitConsole: $(System.DefaultWorkingDirectory)/NUnit.ConsoleRunner/tools/nunit3-console.exe
-    testAssembly: $(System.DefaultWorkingDirectory)/XQA.Android/XQA.Android.dll
-    testRunTitle: TargetFrameworkVersion Tests
-    nunitConsoleExtraArgs: --where "cat == TargetFrameworkTests"
-
-- template: run-nunit-tests.yaml
-  parameters:
-    nunitConsole: $(System.DefaultWorkingDirectory)/NUnit.ConsoleRunner/tools/nunit3-console.exe
-    testAssembly: $(System.DefaultWorkingDirectory)/XQA.Android/XQA.Android.dll
-    testRunTitle: Resource Cache Tests
-    nunitConsoleExtraArgs: --where "cat == ResourceCacheTests"
-
-- template: run-nunit-tests.yaml
-  parameters:
-    nunitConsole: $(System.DefaultWorkingDirectory)/NUnit.ConsoleRunner/tools/nunit3-console.exe
-    testAssembly: $(System.DefaultWorkingDirectory)/XQA.Android/XQA.Android.dll
-    testRunTitle: Property Cache Tests
-    nunitConsoleExtraArgs: --where "cat == PropertyCacheTests"
-
-- template: run-nunit-tests.yaml
-  parameters:
-    nunitConsole: $(System.DefaultWorkingDirectory)/NUnit.ConsoleRunner/tools/nunit3-console.exe
-    testAssembly: $(System.DefaultWorkingDirectory)/XQA.Android/XQA.Android.dll
-    testRunTitle: Code Analysis Tests
-    nunitConsoleExtraArgs: --where "cat == CodeAnalysisTests"
-
-- template: run-nunit-tests.yaml
-  parameters:
-    nunitConsole: $(System.DefaultWorkingDirectory)/NUnit.ConsoleRunner/tools/nunit3-console.exe
-    testAssembly: $(System.DefaultWorkingDirectory)/XQA.Android/XQA.Android.dll
-    testRunTitle: Networking Options Tests
-    nunitConsoleExtraArgs: --where "cat == HttpClientAndTlsProviderPackageTests"
-
-- template: run-nunit-tests.yaml
-  parameters:
-    nunitConsole: $(System.DefaultWorkingDirectory)/NUnit.ConsoleRunner/tools/nunit3-console.exe
-    testAssembly: $(System.DefaultWorkingDirectory)/XQA.Android/XQA.Android.dll
-    testRunTitle: AndroidApiInfo Tests
-    nunitConsoleExtraArgs: --where "cat == AndroidApiInfoTests"
-
-- template: run-nunit-tests.yaml
-  parameters:
-    nunitConsole: $(System.DefaultWorkingDirectory)/NUnit.ConsoleRunner/tools/nunit3-console.exe
-    testAssembly: $(System.DefaultWorkingDirectory)/XQA.Android/XQA.Android.dll
-    testRunTitle: Test Environment Cleanup
-    nunitConsoleExtraArgs: --where "cat == XATestCleanUp"
+- ${{ if eq(parameters.node_id, 1) }}:
+  - powershell: |   
+      if ([Environment]::OSVersion.Platform -eq "Unix") {
+         mono $(System.DefaultWorkingDirectory)/NUnit.ConsoleRunner/tools/nunit3-console.exe $(System.DefaultWorkingDirectory)/XQA.Android/XQA.Android.dll --where "cat == XAUnitTestSetup || cat == XATestPrep || cat == EnvironmentInfo"
+      } else {
+         $(System.DefaultWorkingDirectory)\NUnit.ConsoleRunner\tools\nunit3-console.exe $(System.DefaultWorkingDirectory)\XQA.Android\XQA.Android.dll --where "cat == XAUnitTestSetup || cat == XATestPrep || cat == EnvironmentInfo"
+      }
+    displayName: Test Environment Setup
```

Comment:
We'll want to move this step out and make sure it runs on every node before any other test categories. It's needed to make sure any attached devices/emulators are ready, and to prepare some test sources.

Reply:
Good call, I originally debated on this one for a while.  Moved it up to all nodes.